### PR TITLE
[exporter/opensearch] Send logs to Opensearch

### PR DESCRIPTION
**Pull request by jaehnri**
## Description: <Describe what has changed.>

Implementation of exporter to OpenSearch using opensearch-go library. As of now, this PR was heavily inspired by https://github.com/dbason/opentelemetry-collector-contrib/tree/opensearch-exporter/exporter/opensearchexporter. 

By default, requests sent adhere to the OpenSearch Catalog [schema for logs](https://github.com/opensearch-project/opensearch-catalog/tree/main/schema/observability/logs), but allows users to export using the Elastic Common Schema as well.

This PR also:
- enables users to define the `bulk_action` between `create` and `index`
- enables users to define the logs index without necessarily adhering to the new [index naming conventions](https://github.com/opensearch-project/observability/issues/1405) through the `LogsIndex` config.

## Tracking Issue: 

[23611](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23611)

### Integration
- Successful round-trip to HTTP endpoint,
- Permanent error during round-trip,
- Retryable error response for first request, followed by successful response on retry,
- Two retriable error responses, followed by successful response on second retry.

### Manual
- Authentication using `configtls.TLSSetting` (`ca_file`, `cert_file`, `key_file`)
- Tested in EKS and K3s clusters running [opni](https://github.com/rancher/opni).

### CODE REVIEW

**Comment by jaehnri** [comment]
@Aneurysm9 @bryan-aguilar friendly pinging for review here so the PR isn't closed for inactivity. Thanks!

**Comment by bryan-aguilar** [comment]
Pinging @MitchellGale @MaxKsyunz @YANG-DB as opensearch exporter codeowners.

**Comment by bryan-aguilar** [comment]
i have ran the integration tests, please check their success status and respond to any failures.

**Comment by jaehnri** [comment]
There were some issues regarding the changelog that should be addressed now. Can you run them again, please? Thanks!

**Comment by mikaylathompson** [comment]
This PR would be extremely useful for our use case as the client in the elasticsearch exporter blocks working with opensearch clusters.

If there's anything I can do to help push this PR across the finish line, please let me know. It looks like it's code complete and the tests are passing.

**Comment by sumobrian** [comment]
This PR is exactly what we need to adopt this package for an AWS solutions project. @Aneurysm9 is there anything that is preventing this PR from being merged?

**Comment by bryan-aguilar** [comment]
This PR is still awaiting approval from code owners @sumobrian

**Comment by atoulme** [comment]
Tagging code owners: [@Aneurysm9](https://www.github.com/Aneurysm9), [@MitchellGale](https://www.github.com/MitchellGale), [@MaxKsyunz](https://www.github.com/MaxKsyunz), [@YANG-DB](https://www.github.com/YANG-DB)

**Review by YANG-DB — APPROVED** [review]

**Comment by YANG-DB** [comment]
Thanks for your support
We will push this to get merged ASAP

**Comment by jaehnri** [comment]
@atoulme @YANG-DB @bryan-aguilar rebased to fix conflicts. It should be good to go now.

**Comment by YANG-DB** [comment]
@Aneurysm9 - can you please review
Thanks 👍

File: exporter/opensearchexporter/internal/objmodel/objmodel.go
Comment (Aneurysm9) [inline exporter/opensearchexporter/internal/objmodel/objmodel.go:32]: Can this be scoped to specific statements?  Even if that requires duplication it's preferable to not allowing further unchecked errors to be introduced without discussion.

File: exporter/opensearchexporter/log_bulk_indexer.go
Comment (Aneurysm9) [inline exporter/opensearchexporter/log_bulk_indexer.go:4]: This may also be a candidate for moving to an internal package.

File: exporter/opensearchexporter/sso_model.go
Comment (Aneurysm9) [inline exporter/opensearchexporter/sso_model.go:50]: Resource attributes can be more than just strings.  How does this handle non-string-valued resource attributes?

**Review by Aneurysm9 — APPROVED** [review]
I have reviewed primarily for structure and consistency with collector style and expectations.  The notes here are not blocking as they do not affect the public API, but should be addressed before moving out of the `development` stage.

**Comment by jaehnri** [comment]
For note: I will address the comments in another PR, including a refactor in the bulk indexer structure.

**Comment by YANG-DB** [comment]
Hi
Since the PR is labeled ready for merge - does it need another manual signoff or is it merged automatically??

**Comment by bryan-aguilar** [comment]
An @open-telemetry/collector-contrib-maintainer will come by and merge it when they get a chance. It is not done automatically.

**Comment by bryan-aguilar** [comment]
@jaehnri, please address merge conflicts though as that will prevent a maintainer from merging it in.

**Comment by jaehnri** [comment]
@bryan-aguilar, done it.

**Comment by mx-psi** [comment]
> I have reviewed primarily for structure and consistency with collector style and expectations. The notes here are not blocking as they do not affect the public API, but should be addressed before moving out of the development stage.

@jaehnri can you create a tracking issue for addressing the changes in @Aneurysm9's review? I will merge this now, but please link the issue here once you have created it :)

**Comment by jaehnri** [comment]
@mx-psi, here it is: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28666

**Pull request merged**
